File: src/main/java/pl/training/workshop/m7/s10_booleanparameter/start/BoxOfficeTerminal.java
package pl.training.workshop.m7.s10_booleanparameter.start;

import java.util.Objects;

/**
 * Klient 2: terminal w kasie. customerHasGlasses to dana z formularza, nie flaga sterująca -
 * nie każdy boolean jest zapachem.
 */
public final class BoxOfficeTerminal {
    private final TicketService tickets;

    public BoxOfficeTerminal(TicketService tickets) {
        this.tickets = Objects.requireNonNull(tickets, "tickets");
    }

    public String sell(String title, String format, int seats, boolean customerHasGlasses) {
        return tickets.book(title, format, seats, false, customerHasGlasses);
    }
}

File: csharp/src/Training.Workshop/M7/S10BooleanParameter/Start/BoxOfficeTerminal.cs
namespace Training.Workshop.M7.S10BooleanParameter.Start;

/// <summary>
/// Klient 2: terminal w kasie. customerHasGlasses to dana z formularza, nie flaga sterująca -
/// nie każdy bool jest zapachem.
/// </summary>
public sealed class BoxOfficeTerminal
{
    private readonly TicketService _tickets;

    public BoxOfficeTerminal(TicketService tickets)
    {
        ArgumentNullException.ThrowIfNull(tickets);
        _tickets = tickets;
    }

    public string Sell(string title, string format, int seats, bool customerHasGlasses)
    {
        return _tickets.Book(title, format, seats, false, customerHasGlasses);
    }
}
